arithmetization/src/main/java/net/consensys/linea/zktracer/module/ec_data/EcDataOperation.java

```diff
@@ -0,0 +1,348 @@
+    }
+  }
+
+  private static int ecTypeToNRows(int ecType, final Bytes input) {
+    return switch (ecType) {
+      case EC_RECOVER, EC_ADD -> 8;
+      case EC_MUL -> 6;
+      case EC_PAIRING -> input.size() / 16;
+      default -> throw new IllegalArgumentException("invalid EC type");
+    };
+  }
+
+  private static boolean isOnG2(Bytes arg) {
+    return true; // TODO:
+  }
+
+  private static boolean isInfinity() {
+    return false; // TODO:
+  }
+
+  private EcDataOperation(
+      Wcp wcp, Ext ext, int contextNumber, int previousContextNumber, int ecType, Bytes input) {
+    Preconditions.checkArgument(EC_TYPES.contains(ecType), "invalid EC type");
+
+    final int nRows = ecTypeToNRows(ecType, input);
+    final int minInputLength = ecType == EC_MUL ? 96 : 128;
+    if (input.size() < minInputLength) {
+      this.input = leftPadTo(input, minInputLength);
+    } else {
+      this.input = input;
+    }
+    this.contextNumber = contextNumber;
+    this.contextNumberDelta = contextNumber - previousContextNumber;
+    this.ecType = ecType;
+    this.nRows = nRows;
+    this.thisIsNotOnG2Index = -1;
+    if (ecType == EC_PAIRING) {
+      this.nPairings = input.size() / 192;
+    } else {
+      this.nPairings = 0;
+    }
+
+    this.comparisons = repeat(false, nRows / 2);
+    this.equalities = repeat(false, nRows);
+    this.squares = repeat(Bytes.EMPTY, nRows / 2);
+    this.cubes = repeat(Bytes.EMPTY, nRows / 2);
+    this.limbs = repeat(Bytes.EMPTY, nRows / 2);
+
+    this.wcpArg1 = repeat(Bytes.EMPTY, nRows);
+    this.wcpArg2 = repeat(Bytes.EMPTY, nRows);
+    this.wcpInst = repeat(OpCode.INVALID, nRows);
+    this.wcpRes = repeat(false, nRows);
+
+    this.extArg1 = repeat(Bytes.EMPTY, nRows);
+    this.extArg2 = repeat(Bytes.EMPTY, nRows);
+    this.extArg3 = repeat(Bytes.EMPTY, nRows);
+    this.extInst = repeat(OpCode.INVALID, nRows);
+    this.extRes = repeat(Bytes.EMPTY, nRows);
+    ;
+
+    this.wcp = wcp;
+    this.ext = ext;
+  }
+
+  public static EcDataOperation of(
+      Wcp wcp,
+      Ext ext,
+      final Address to,
+      Bytes input,
+      int currentContextNumber,
+      int previousContextNumber) {
+    final int ecType = addressToEcType(to);
+
+    EcDataOperation r =
+        new EcDataOperation(wcp, ext, currentContextNumber, previousContextNumber, ecType, input);
+    switch (ecType) {
+      case EC_RECOVER -> r.handleRecover();
+      case EC_ADD -> r.handleAdd();
+      case EC_MUL -> r.handleMul();
+      case EC_PAIRING -> r.handlePairing();
+    }
+    return r;
+  }
+
+  private boolean preliminaryChecksPassed() {
+    return this.hurdleSwitchOffRow == -1;
+  }
+
+  private boolean callWcp(int i, OpCode inst, Bytes arg1, Bytes arg2) {
+    final boolean r =
+        switch (inst) {
+          case LT -> this.wcp.callLT(arg1, arg2);
+          case EQ -> this.wcp.callEQ(arg1, arg2);
+          default -> throw new IllegalStateException("Unexpected value: " + inst);
```

Comment:
it's more illegalArgument, we could call wcp for slt, sgt, etc

Reply:
We are in `EcDataOperation` here; if it happens someone typed the wrong opcode somewhere in the module.